OBSDOCS-996: Create a table listing and describing all RBAC permissio… (#75852)
openshift/openshift-docs: modules/monitoring-granting-users-permission-to-monitor-user-defined-projects.adoc

[id="granting-users-permission-to-monitor-user-defined-projects_{context}"]
= Granting users permission to monitor user-defined projects

As a cluster administrator, you can monitor all core {product-title} and user-defined projects.

You can also grant developers and other users different permissions:

* To monitor user-defined projects.
* To configure the components that monitor user-defined projects.
* To configure alert routing for user-defined projects.

You can grant the permissions by assigning one of the following monitoring roles:

|===

> |Cluster role name |Description

Reviewer: ```suggestion
|Role name |Description
```
reason, see from https://github.com/openshift/openshift-docs/pull/75852/files#r1599953870, `user-workload-monitoring-config-edit` is role, not cluster role

> |`monitoring-rules-view` | Users with this role have read access to `PrometheusRule` custom resources for a user-defined project.

Author: That would be definitely helpful to mention, thank you!

> |`monitoring-rules-view` | Users with this role have read access to `PrometheusRule` custom resources for a user-defined project. They can also view the *Alerts* page in the *Developer* perspective of the {product-title} web console.

Reviewer: ```suggestion
|`monitoring-rules-view` | Users with this cluster role have read access to `PrometheusRule` custom resources for a user-defined project. They can also view the *Alerts* page in the *Developer* perspective of the {product-title} web console.
```
reason
```
$ oc get clusterrole monitoring-rules-view
NAME                    CREATED AT
monitoring-rules-view   2024-05-14T03:41:36Z
```

> |`monitoring-rules-edit` | Users with this role can create, modify, and delete `PrometheusRule` custom resources for a user-defined project. They can also silence alerts.

Reviewer: ```suggestion
|`monitoring-rules-edit` | Users with this role can create, modify, and delete `PrometheusRule` custom resources for a user-defined project. They can create and silence alerts in the `Developer` perspective in the web console
```
see from: https://docs.openshift.com/container-platform/4.15/observability/monitoring/managing-alerts.html

Author: Hello @juzhao, I noticed that in the original list we did not include `monitoring-alertmanager-edit` , I could add that role to the table as well, WDYT?

Reviewer: @eromanova97 monitoring-alertmanager-edit and cluster-monitoring-view,  thery are used to silence alert, this PR does not contain silence alert. @simonpasquier WDYT?

Reviewer 2: monitoring-alertmanager-edit and cluster-monitoring-view roles are specific to platform components while this PR is about user-defined monitoring. We should be more explicit about these roles but probably not in this PR?

Author: Yes, I was thinking about this as well. Maybe for now I will just include the user-defined list as it was intended. We can raise a new issue to create something similar for other roles. :+1: thank you for your inputs!

> |`monitoring-edit` | Users with this role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.

Reviewer: ```suggestion
|`monitoring-edit` | Users with this cluster role have the same privileges as users with the `monitoring-rules-edit` cluster role. Additionally, users can create, modify, and delete `ServiceMonitor` and `PodMonitor` resources to scrape metrics from services and pods.
```

```
$ oc get clusterrole monitoring-edit
NAME              CREATED AT
monitoring-edit   2024-05-14T03:41:36Z
```

> |`user-workload-monitoring-config-edit` | This cluster role is given in the `openshift-user-workload-monitoring` project. Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring.

Reviewer: ```suggestion
|`user-workload-monitoring-config-edit` | This role is given in the `openshift-user-workload-monitoring` project. Users with this role can edit the `user-workload-monitoring-config` `ConfigMap` object to configure Prometheus, Prometheus Operator, Alertmanager, and Thanos Ruler for user-defined workload monitoring.
```

```
$ oc -n openshift-user-workload-monitoring get role user-workload-monitoring-config-edit
NAME                                   CREATED AT
user-workload-monitoring-config-edit   2024-05-14T03:41:37Z
```

> |`alert-routing-edit` | Users with this role can create, update, and delete `AlertmanagerConfig` custom resources for a user-defined project.

Reviewer: ```suggestion
|`alert-routing-edit` | Users with this cluster role can create, update, and delete `AlertmanagerConfig` custom resources for a user-defined project.
```
```
$ oc get clusterrole alert-routing-edit
NAME                 CREATED AT
alert-routing-edit   2024-05-14T03:41:37Z
```
